src/lib.rs

```diff
@@ -1092,10 +1092,8 @@ impl<E, DEV> Mpu9250<DEV, Dmp> where DEV: Device<Error = E>
 
         let mut buffer: [u8; 33] = [0; 33];
         let read = self.read_fifo(&mut buffer[..self.packet_size + 1])?;
-        if read == -(self.packet_size as isize) {
+        if read < 0 {
             return Err(Error::DmpDataNotReady);
-        } else if read != 0 {
-            return Err(Error::DmpDataInvalid);
```

Comment:
Here, there is a state were something wrong happened and the fifo need to be reset. If you delete this mechanism the function will not detect that the fifo is in an error state. There is an error state because we expect to receive data by N bytes chunk but we received a different amount. Therefore, we don't know what we are reading and don't have any clue to parse it.

Reply:
Are you sure the data in the fifo is always a multiple of the package size? I actually don’t think so, that is why the reference implementation is checking the magnitude of the quaternion to detect a faulty transmission.

Reply:
I’ve just tested it again with the proposed change and it gets into the error state after a short amount of time, repeatable. So I’m pretty sure it is not always a multiple of packages size.

We could introduce the quaternion magnitude check (later?), but it should be optional.

Reply:
Ok I never add this issue. I m interested to know what is your configuration.

Reply:
I'm reading the fifo only in about 25 Hz intervals, you could try to slow it down on your board.